src/events/SDL_events.c

```diff
@@ -973,7 +1012,7 @@ SDL_WaitEventTimeout(SDL_Event * event, int timeout)
         expiration = start + timeout;
     }
 
-    if (timeout != 0 && _this && _this->WaitEventTimeout && _this->SendWakeupEvent && !SDL_events_need_polling()) {
+    if (_this && _this->WaitEventTimeout && _this->SendWakeupEvent && !SDL_events_need_polling()) {
```

Comment:
I don't think we want to remove the `timeout != 0` check here. It wasn't part of the poll sentinel stuff.

I added it in b992b915e520fed9e8558c9e6d8ca66370c04e64

Reply:
timeout == 0 will never hit this code now, but we can leave it in for clarity, if you want.

Reply:
Ah, right. You could add an `SDL_assert(timeout != 0)` and comment if you want. That's probably clearer than leaving the superfluous `timeout != 0` check in there.